gef-ui/server/controllers/exporters-address/index.js

```diff
@@ -43,29 +44,39 @@ const validateExportersAddress = async (req, res) => {
     }
 
     if (correspondence === 'true' && _isEmpty(postcode)) {
-      correspondenceError = {
+      correspondenceError.push({
         errRef: 'postcode',
         errMsg: 'Enter a postcode',
-      };
+      });
     }
 
-    return res.render('partials/exporters-address.njk', {
-      errors: validationErrorHandler(correspondenceError),
-      companyName: details.companyName,
-      registeredAddress,
-      applicationId,
-      correspondence,
-    });
+    // Only fetch addresses if there are no validation errors
+    if (correspondenceError.length === 0) {
+      addresses = await api.getAddressesByPostcode(postcode);
+    }
 
-    // return res.redirect('exporters-address');
-  } catch (err) {
-    // Validation errors
-    if (err.status === 422) {
-      return res.render('partials/companies-house.njk', {
-        errors: validationErrorHandler(err.data),
+    // Check for validation errors
+    if (addresses && addresses.status === 422) {
```

Comment:
Same question as above about whether only a 422 status gets returned on error. Should we assume anything other than 200 is error?